pkg/upgrade/drain.go

```diff
@@ -20,12 +20,14 @@ import (
 )
 
 func (u *simpleUpgrader) drain(ctx context.Context, cs *api.OpenShiftManagedCluster, role api.AgentPoolProfileRole, nodeName string) error {
-	kc, err := managedcluster.ClientsetFromV1Config(cs.Config.AdminKubeconfig)
-	if err != nil {
-		return err
+	var err error
+	if u.kubeclient == nil {
+		u.kubeclient, err = managedcluster.ClientsetFromV1Config(cs.Config.AdminKubeconfig)
```

Comment:
I know that you are not changing this but I don't think `drain` should need access to admin kubeconfig in order to perform the drain. Can you open a follow up issue to scope this down?

Reply:
https://github.com/openshift/openshift-azure/issues/609